airflow/utils/log/s3_task_handler.py

```diff
@@ -104,17 +105,19 @@ def _read(self, ti, try_number, metadata=None):
         log_relative_path = self._render_filename(ti, try_number)
         remote_loc = os.path.join(self.remote_base, log_relative_path)
 
+        old_remote_loc = self._gen_old_remote(remote_loc)
+
         if self.s3_log_exists(remote_loc):
-            # If S3 remote file exists, we do not fetch logs from task instance
-            # local machine even if there are errors reading remote logs, as
-            # returned remote_log will contain error messages.
-            remote_log = self.s3_read(remote_loc, return_error=True)
-            log = '*** Reading remote log from {}.\n{}\n'.format(
-                remote_loc, remote_log)
-            return log, {'end_of_log': True}
+            pass
+        elif self.s3_log_exists(old_remote_loc):
+            remote_loc = old_remote_loc
```

Comment:
@cmiller01 Consider prepending a message to `log` in here indicating that the remote location needed to be overridden.